### PR TITLE
add copperhead doctor command for necessary diagnosis

### LINKED ISSUE #78: Add optional install.sh bootstrap script for easier first-time setup
### The problem

New users currently need to install KiCad separately and ensure kicad-cli is on PATH before copperhead check can run. That creates friction for first-time contributors, especially on macOS/Linux.

### Proposed behavior

Add an optional install.sh bootstrap script that:

* Checks for Node.js.
* Checks whether kicad-cli is available.
* Detects the OS/package manager when possible.
* Installs copperhead or guides the user through the setup.
* Verifies the final setup with kicad-cli version and a basic copperhead check.

### Why this helps

* Reduces setup friction for new users.
* Gives a one-command onboarding path for macOS/Linux users.
* Keeps the current npm-based workflow intact for advanced users.
* Provides a clearer entry point than asking people to manually piece together prerequisites.

### Suggested scope

* Start with macOS and Linux support.
* Keep it idempotent and conservative.
* Print clear next steps if automatic installation is not possible.
* Avoid silent changes to shell config without user consent.

### Acceptance criteria

* Script detects missing prerequisites and prints actionable guidance.
* Script succeeds on a clean machine where dependencies are available.
* Script exits safely with clear messaging when auto-install is not supported.
* README includes usage instructions and supported platforms.

### Possible follow-up

* Add a copperhead doctor command that checks prerequisites and prints a setup report.

### Alternatives you considered

Workaround was installing based on commands given as part installation failure.

### Effect on the invariants

No effect doesn't touch OpenSpec or do any mutations. 

### Before submitting

- [x] I searched existing issues and the roadmap, and this is not already tracked.

### PR DESCRIPTION
**Pull request by AniketR10**
## What

Adds `copperhead doctor`: a fast, LLM-free, network-free environment preflight command that checks whether the machine can actually run copperhead before you start a real command, and prints an actionable hint for each thing that's missing.

Also adds [install.sh](install.sh), a conservative bootstrap installer for macOS and Linux that checks the prerequisites, installs copperhead (globally via npm, or built from source inside a checkout), and verifies the result with `copperhead doctor`.

Closes #78

## Why

`check` verifies the design (ERC/DRC/drift), but it is contractually LLM-free, so it never looks at the model provider. Today the only way to discover a missing `OPENAI_API_KEY`/`ANTHROPIC_API_KEY` is to start a real `do`/`create` run and let it fail mid-flight, several minutes and several turns in. `doctor` surfaces that up front, in under a second, alongside node/kicad-cli/git presence.

`install.sh` closes the loop from the other side (#78): new users get a one-command onboarding path that ends by running `doctor`, while the npm-based workflow stays intact for everyone else.

## Design

`src/commands/doctor.ts` exports a pure `runDoctor()` with five checks: node version, kicad-cli, git, the resolved model's provider credential, and an informational project-config summary. Decisions worth calling out:

- **Credential resolution mirrors a real run exactly.** `runDoctor` calls the same `resolveModel()` (`--model` > `COPPERHEAD_MODEL` > config > available key) used by `do`/`create`, then routes the result to a provider the same way `makeProvider()` does (`codex`/`claude-code` prefixes need no key and report `info`; everything else needs `OPENAI_API_KEY` or `ANTHROPIC_API_KEY`). So `doctor` never gives an answer a real run would contradict.
- **Every probe fails soft.** Unlike every other command, `doctor` does not call `kicadCliVersion()` at the top and let it throw; a missing binary becomes a reported `[FAIL]` row with an install hint, so the report is always complete. Probes are injectable (`DoctorDeps`) so tests never depend on the host machine.
- **Output is column-aligned, width-aware, and color-capable.** Status tags are padded outside the brackets, and detail/hint text wraps to the terminal width with continuation lines aligned to their column. Colors (green/red/cyan tags, dim hints) are plain ANSI with no dependency, and strictly opt-in at the CLI layer: only on a real TTY with `NO_COLOR` unset, so piped output, CI, `--json`, and tests always see plain text. The no-model error (shared with `do`/`sync` via `resolveModel`) was shortened to point at the configuration docs instead of inlining every option.
- **`install.sh` never escalates.** It never runs sudo and never edits shell config; whenever it cannot act safely it prints the exact command for the user to run. Non-interactive shells default to no, so `curl | bash` without `--yes` can only report, never install. It probes for `doctor` via the help listing (not the version, which is unchanged in this PR), so it degrades gracefully against older installed releases.

Doesn't touch the agent loop, tool gating, or KiCad file editing, so none of the safety invariants are in play here.

## Spec / OpenSpec

No spec-level behavior changed (no AC, no invariant, no agent-loop/tool/KiCad-layer touch). Note for the reviewer: `SPEC.md` §3 ("CLI surface") enumerates the commands and does not yet list `doctor`; flagging rather than filing an OpenSpec change, since this is purely additive tooling with no acceptance criteria of its own. Happy to add a short §3 mention if preferred.

### Automated test status

| Check | Command | Status |
| --- | --- | --- |
| Typecheck | `npm run typecheck` | pass |
| Build | `npm run build` | pass |
| Unit + offline | `npm test` | pass: 359 passed, 17 skipped, 0 failed (post-merge with main's cursor provider) |
| Live-LLM (opt-in) | n/a | doctor makes no LLM/network calls; no live-provider variable gates it |

New: `test/doctor.test.ts` (15 tests): all-green case, missing provider key, missing kicad-cli and missing git (soft-fail, not thrown), malformed and unreadable config, saved-login providers (`codex`, `cursor`, `claude-code`) report `info` not `fail`, old-node-major failure, `checkCredential` prefix routing, wired-project summary, no-model hint dedup, and `formatDoctor` footer, width-wrapping/alignment, and color-mode (stripping ANSI codes yields exactly the plain output, so color can never change layout).

### Manual test log (required)

- Sandbox variant used: `edit` (for the project-check row), plus standalone runs (doctor has no LLM-driven path, so `create`/`edit`'s agent-loop setup doesn't apply to most of it)
- Commands run and outcome:

```text
$ copperhead doctor
  [ok]   node      v24.18.0 (>= 20)
  [ok]   kicad-cli 10.0.4
  [ok]   git       2.34.1
  [FAIL] provider  no model configured
         hint: pass --model, set COPPERHEAD_MODEL, or export an API key; see
               https://docs.copperhead.sh/reference/configuration/
  [info] project   no .copperhead/config.json (run `copperhead init` to
                   scaffold)
not ready: fix the [FAIL] items above
(exit 1; on a TTY the tags render green/red/cyan and hints dim, NO_COLOR disables)

$ OPENAI_API_KEY=sk-test-x copperhead doctor --model gpt-5
  [ok]   provider  gpt-5 -> openai: OPENAI_API_KEY set
ready
(exit 0)

$ copperhead doctor --model claude-code
  [info] provider  claude-code -> claude-code: uses Claude Code login (not verified offline)

$ copperhead doctor --json
{ "ok": false, "checks": [ { "name": "node", "status": "ok", ... } ] }
(shape unchanged; JSON output never carries ANSI codes)

$ ./install.sh --check-only
copperhead bootstrap (linux, package manager: apt-get)
1/4 Node.js            ok    node v24.18.0 (>= 20)
2/4 git                ok    git version 2.34.1
3/4 KiCad (kicad-cli)  ok    kicad-cli 10.0.4
4/4 copperhead         ok    copperhead 0.7.0
(exit 0; idempotent, installs nothing on a ready machine)

$ bash -n install.sh
(syntax clean; verification block captures doctor output, which correctly
drops colors when piped and re-indents cleanly)
```

Confirmed via `readlink -f "$(which copperhead)"` that the global binary resolved to this branch's `dist/cli.js` throughout.

Reviewer note: `package.json` version is still 0.7.0. Since older releases are distinguished from this one only by the help-listing probe, the release that ships `doctor` should bump the version so `npm install -g copperhead@latest` actually upgrades.

## Docs

- [README.md](README.md): added `copperhead doctor` to the command list, and a "Bootstrap script" section for `install.sh`.
- [docs/src/content/docs/reference/cli.md](docs/src/content/docs/reference/cli.md): new `## copperhead doctor` reference section (synopsis, checks table, exit codes, `--json` shape).
- [docs/src/content/docs/reference/configuration.md](docs/src/content/docs/reference/configuration.md): documented `NO_COLOR` in the environment variables table.

---

## Invariant checklist

- [x] N/A: **Spec-gated in**: `doctor` adds no tool and never touches `edit_file`/`write_file` gating.
- [x] N/A: **Verification-gated out**: `doctor` performs no mutation; nothing to verify or roll back.
- [x] **`check`/`verify` stays LLM-free and network-free**: unchanged; `doctor` is a separate command, itself also LLM-free and network-free (credential check is presence-only, no auth ping).
- [x] N/A: **No sexp serialization**: `doctor` doesn't touch `src/kicad/` parsing or KiCad files.
- [x] N/A: **Sync-obligations ledger**: `doctor` never opens a `RunContext`/ledger.
- [x] N/A: **Secrets**: `doctor` never logs or stores a key value, only whether the env var is set (presence check). `install.sh` likewise never reads or prints credentials.
- [x] N/A: **Spec coherence**: no AC or invariant changed (see Spec/OpenSpec note above re: `SPEC.md` §3 listing).

## Summary by CodeRabbit

* **New Features**
  * Added `copperhead doctor`: an offline, non-LLM, non-network preflight that runs readiness checks and prints a per-check report.
  * Supports human output and `--json`, with `0` when ready and `1` otherwise.
* **Documentation**
  * Documented `copperhead doctor` in the CLI reference, including `--model`/`--interactive` scope and `--json` output.
  * Updated installation and configuration docs, including `NO_COLOR` behavior and clearer `--model` guidance when missing.
* **Tests**
  * Added/expanded coverage for doctor readiness logic, credential routing, config edge cases, and report formatting (including ANSI/no-ANSI output).

### CODE REVIEW

File: src/commands/doctor.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  let configError: DoctorCheck | undefined;
+  try {
+    config = await loadConfig(opts.repoRoot);
+  } catch (err) {
+    config = FALLBACK_CONFIG;
+    configError = {
+      name: 'project',
+      status: 'fail',
+      detail: `.copperhead/config.json is malformed: ${(err as Error).message}`,
+      hint: 'fix or delete .copperhead/config.json (rerun `copperhead init`/`copperhead create` to regenerate it).',
```
[the remark this answers is not in the document]
Comment (AniketR10) [reply src/commands/doctor.ts]: resolved

**Comment by animesh-chouhan** [comment]
**Automated pr-review pass** (Claude Code `/pr-review 96`): findings below are machine-generated and verified against the working tree; this comment is not a formal review and does not affect merge gating.

## Verdict: approve with nits

`copperhead doctor` is a clean, well-isolated addition: the credential routing genuinely mirrors `makeProvider`'s prefix order (codex, then claude-code, then the `claude` prefix, then OpenAI fallback), `FALLBACK_CONFIG` matches `loadConfig`'s no-file shape exactly, and every probe fails soft as described. CI is green, the suite passes, and no repo invariant is touched. The findings are all low severity: one small divergence from the PR's own "doctor never contradicts a real run" claim, one readiness check a real run enforces that doctor skips, the untested branches behind the 81.1% diff coverage, and the already-flagged SPEC.md §3 listing.

## Metrics

```
lines: +408 / -1 (net 407) across 5 files  ·  src +244/-0, test +139/-0, docs +25/-1, spec +0/-0, other +0/-0
new vs net: 2 new files (1 in src), 3 modified; 228 new src lines (the net-new surface)
tests: +139 test lines; suite not measured -> 337/14/0 (pass/skip/fail, base -> head)
diff coverage: 81.1% of changed executable src lines exercised (137/169, measured via vitest v8); uncovered: src/cli.ts:9,125-130,134-138, src/commands/doctor.ts:81-87,142-148,160-162,222-224
```

Method: areas from `git diff --numstat` against merge base `887346b`, generated files excluded; suite and coverage from an actual `npx vitest run --coverage` on the checked-out head, intersected with the changed lines from `git diff -U0`. Base suite: not measured (base CI on main is the reference; `--base-tests` not run). PR state from `gh pr view`: `mergeable=MERGEABLE`, not a draft; all four checks green (`test` 1m1s pass).

## Findings

**1. [low] `doctor` reports a ready `info` for an empty saved-login model override that a real run rejects** — [doctor.ts:98](src/commands/doctor.ts#L98) and [doctor.ts:105](src/commands/doctor.ts#L105). `checkCredential` routes on `model.startsWith('codex:')` / `startsWith('claude-code:')` without checking the part after the colon, but `makeProvider` ([loop.ts:78](src/agent/loop.ts#L78)) throws `codex model override cannot be empty` for exactly that input. Failure scenario: `copperhead doctor --model codex:` prints `[info] provider` and exits 0 ("ready"), then `copperhead do --model codex:` fails immediately — contradicting the PR's design claim that doctor never gives an answer a real run would contradict. Fix: in both saved-login branches, report `fail` with the same wording as `makeProvider` when the `:` suffix is empty (one condition per branch, plus a `checkCredential('codex:', {})` assertion in the prefix-routing test).

**2. [low] doctor omits the disk-space preflight that `create` enforces** — [doctor.ts:172](src/commands/doctor.ts#L172) (the check list). `runCreate` refuses to start below 2 GiB free via `assertDiskSpace` ([preflight.ts](src/util/preflight.ts)), so a machine doctor calls "ready" can still be rejected by the very next command. Failure scenario: free disk at 1 GiB, `copperhead doctor` exits 0, `copperhead create` throws `PreflightError` before doing anything. Fix: add a `disk` row reusing `freeDiskBytes` (as an injectable probe like the others; `null` reading stays `info`, below-threshold becomes `fail` or at least a warning `info`).

**3. [low] Untested surface behind the 81.1% diff coverage** — each uncovered range was traced by hand and none hides a defect, but four are asserted nowhere: the `gitCheck` failure branch ([doctor.ts:81-87](src/commands/doctor.ts#L81-L87), symmetric twin of the tested `kicadCheck` one); the `providerCheck` no-model-configured catch ([doctor.ts:142-148](src/commands/doctor.ts#L142-L148)), whose hint is the whole UX of the PR's headline scenario and is only exercised in the manual log; the wired-project branch ([doctor.ts:160-162](src/commands/doctor.ts#L160-L162), the `schematic … · board …` summary); and `formatDoctor`'s per-row and hint rendering ([doctor.ts:222-224](src/commands/doctor.ts#L222-L224)) — both `formatDoctor` tests pass `checks: []`, so the row/hint lines every user reads are never asserted. The `src/cli.ts:125-138` action block is also uncovered, consistent with this repo's convention of not testing commander wiring. Fix: one `runDoctor` test with no model anywhere plus a `formatDoctor` call on a report containing a hint would close the meaningful gaps.

**4. [low] SPEC.md §3 (CLI surface) does not list `doctor`** — [SPEC.md:235](openspec/specs/SPEC.md#L235). The author flagged this in the description rather than diverging silently, and §3 is already slightly stale (it predates `export bom`, which README lists), so this is a nit, not a blocker: recommend taking the offered one-line §3 mention in this PR so the drift doesn't compound. No OpenSpec change needed; no AC or invariant moves.

Note (not a finding): the real `kicadCliVersion` / `execFile git --version` probe paths are mock-only in the suite by design (`DoctorDeps` injection); the PR's manual test log exercises them live, which is adequate for presence probes.

## Invariant checks performed

All six checked explicitly; all pass:

- **Spec-gated in**: no tool-list change; `doctor` never touches `edit_file`/`write_file` gating. Pass.
- **Verification-gated out**: `doctor` performs no mutation; the loop's repair/rollback path is untouched. Pass.
- **`check`/`verify` LLM-free and network-free**: `src/commands/check.ts` and its imports unchanged; `doctor` itself is also offline (local `execFile` probes and env presence checks only, verified in [doctor.ts](src/commands/doctor.ts) — no provider import, no fetch). Pass.
- **No sexp serialization**: `src/kicad/` untouched except the read-only `kicadCliVersion` import. Pass.
- **Sync-obligations ledger**: no `RunContext`/ledger interaction. Pass.
- **Secrets**: only env-var presence is reported, never a value; no transcript writing. Pass.

Spec coherence: no delta-spec or `tasks.md` change was needed (nothing spec-level moved); `openspec validate` not run since no planning artifact changed. See finding 4 for the §3 listing.

**Comment by animesh-chouhan** [comment]
Screenshot of updated doctor implimage
